Content.Server/Atmos/Piping/EntitySystems/AtmosUnsafeUnanchorSystem.cs

```diff
@@ -48,56 +46,54 @@ private void OnUnanchorAttempt(EntityUid uid, AtmosUnsafeUnanchorComponent compo
             }
         }
 
-        private void OnBeforeUnanchored(EntityUid uid, AtmosUnsafeUnanchorComponent component, BeforeUnanchoredEvent args)
+        // When unanchoring a pipe, leak the gas that was inside the pipe element.
+        // At this point the pipe has been scheduled to be removed from the group, but that won't happen until the next Update() call in NodeGroupSystem,
+        // so we have to force an update.
+        // This way the gas inside other connected pipes stays unchanged, while the removed pipe is completely emptied.
+        private void OnUserUnanchored(EntityUid uid, AtmosUnsafeUnanchorComponent component, UserUnanchoredEvent args)
         {
             if (component.Enabled)
+            {
+                _group.ForceUpdate();
                 LeakGas(uid);
+            }
         }
 
         private void OnBreak(EntityUid uid, AtmosUnsafeUnanchorComponent component, BreakageEventArgs args)
         {
-            LeakGas(uid);
+            LeakGas(uid, false);
             // Can't use DoActsBehavior["Destruction"] in the same trigger because that would prevent us
             // from leaking. So we make up for this by queueing deletion here.
             QueueDel(uid);
```

Comment:
Here the deletion is queued but not yet done -- so is the copy needed?

Reply:
What happens without the copy is this:
- The pipe is broken
- We leak gas from the pipe group, to which it is still connected until next tick
- This will reduce the amount of gas in the pipe group accordingly
- The pipe is deleted
- This code in PipeNet.cs runs, reducing the gas amount a second time
![grafik](https://github.com/user-attachments/assets/3642c510-d18f-46c3-b519-05e40f2cebd1)
- this would lower the pressure in the remaining pipes, which should not happen

An alternative would be to do a group update here as well, but that would be really problematic performance-wise. for example when the nuke explodes a lot of pipes will break at once.

Reply:
Understood, thanks.